### ISSUE TITLE
Missing component metadata for remap runtime failures

### ISSUE THREAD

**Issue by debugmiller** [vectordotdev/vector#11871]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Use Cases

We recently had a remap transform that had a runtime error. However the error message and the  [processing_errors_total metric](https://vector.dev/docs/reference/configuration/transforms/remap/#processing_errors_total) were missing component metadata, making this difficult to understand which of our transforms had the problem.

The log line was
```
2022-03-16T17:19:26.297447Z  WARN vector::internal_events::remap: Mapping failed with event. error="function call error for \"to_timestamp\" at (230:269): unable to coerce \"null\" into \"timestamp\"" internal_log_rate_secs=30
```

You'll notice in the above message that no component metadata was included, even though in my testing I see it in other transform runtime errors.

The metric that was generated from this event (in prometheus format) was:

```
# HELP vector_processing_errors_total processing_errors_total
# TYPE vector_processing_errors_total counter
vector_processing_errors_total{error_type="failed_mapping"} 1 1647457459096
```

The  [docs indicate](https://vector.dev/docs/reference/configuration/transforms/remap/#processing_errors_total) that this metric should contain component metadata.

### Version

vector 0.19.0 (x86_64-unknown-linux-gnu da60b55 2021-12-28)

**Comment by debugmiller** [comment]
My mistake, this should have `type: bug`

**Comment by jszwedko** [comment]
> My mistake, this should have `type: bug`

Hi @debugmiller ,

I think this bug was fixed in 0.20. Can you try that?

**Comment by debugmiller** [comment]
yes this is fixed in 0.20
